AutoCollection/HttpRequests.ts

```diff
@@ -232,21 +232,40 @@ class AutoCollectHttpRequests {
         }
     }
 
+    private static addResponseCorrelationIdHeaderFromString(client: TelemetryClient, response: http.ServerResponse, correlationHeader: string) {
+        const components = correlationHeader.split(",");
+        const key = `${RequestResponseHeaders.requestContextSourceKey}=`;
+        const found = components.some((value) => value.substring(0,key.length) === key)
+
+        if (!found) {
+            response.setHeader(
+                RequestResponseHeaders.requestContextHeader,
+                `${correlationHeader},${RequestResponseHeaders.requestContextSourceKey}=${client.config.correlationId}`);
+        }
+    }
+
     /**
      * Add the target correlationId to the response headers, if not already provided.
      */
     private static addResponseCorrelationIdHeader(client: TelemetryClient, response:http.ServerResponse) {
         if (client.config && client.config.correlationId &&
             response.getHeader && response.setHeader && !(<any>response).headersSent) {
-            const correlationHeader = response.getHeader(RequestResponseHeaders.requestContextHeader);
-            if (correlationHeader) {
-                const components = correlationHeader.split(",");
-                const key = `${RequestResponseHeaders.requestContextSourceKey}=`;
-                if (!components.some((value) => value.substring(0,key.length) === key)) {
-                    response.setHeader(
-                        RequestResponseHeaders.requestContextHeader,
-                        `${correlationHeader},${RequestResponseHeaders.requestContextSourceKey}=${client.config.correlationId}`);
+            const correlationHeader = <any>response.getHeader(RequestResponseHeaders.requestContextHeader);
+            let header: string;
+            if (typeof correlationHeader === "string") { // string
```

Comment:
This "safe read" logic should probably be made a util since it's needed in two files